lnwallet/wallet.go

```diff
@@ -685,9 +685,16 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 		return
 	}
 
+	// We need to avoid enforcing reserved value in the middle of PSBT
+	// funding because some of the following steps may add UTXOs funding
+	// the on-chain wallet.
+	// The enforcement still happens at the last step - in PsbtFundingVerify
+	enforceNewReservedValue := false
+
 	// If no chanFunder was provided, then we'll assume the default
 	// assembler, which is backed by the wallet's internal coin selection.
 	if req.ChanFunder == nil {
+		enforceNewReservedValue = true
```

Comment:
There are three types of `ChanFunder`: PSBT, canned (used for Pool) and the wallet assembler. We only want to skip this check for the wallet assembler. So I think we should use the default value `true` and only disable it for the PSBT funder (with a type check).

Reply:
I'm not sure I understood you correctly, so I added another commit. Is that what you had in mind?

Reply:
Yes, this is what I meant. But it should be done in the same commit.